### PR TITLE
feat: resumable uploads

### PR DESCRIPTION
**Pull request by suraj-turing**

### CODE REVIEW

File: src/renderer/App.tsx [hunk 1]
```diff
@@ -40,14 +45,106 @@ function AppRoutes() {
...
+      }
+
+      try {
+        const decodedToken: any = token ? jwtDecode(token) : null;
+        const expiryTime = decodedToken ? decodedToken.exp * 1000 : 0;
+        const currentTime = Date.now();
+
+        if (!token || expiryTime < currentTime) {
+          // Token is expired or not available
+          const response = await axios.post(`${BACKEND_URL}/refresh-token`, {
```
Comment (dsasidhar) [inline src/renderer/App.tsx:83]: Why not use axios interceptor on failure, why check expiry manually? This is not guaranteed to run we will run into issues this way.

File: src/renderer/App.tsx [hunk 2]
```diff
@@ -40,14 +45,106 @@ function AppRoutes() {
...
+              <AppHeader />
+              <BusyOverlay />
+              <NavigationButton />
+              <AppRoutes />
+              <Logout />
+            </div>
+          ) : (
+            <div className="h-full flex flex-col">
+              <AppHeader />
+              <div className="flex-grow flex items-center justify-center">
```
Comment (dsasidhar) [inline src/renderer/App.tsx:123]: nit: Move this html to a component, to make this more readable.

File: src/renderer/components/Logout.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      "Log out from Trace? Make sure you're not recording or processing any videos. Only log out if you've saved all your recordings.",
+      {
+        type: DialogType.Confirmation,
+        buttons: ['Yes, Log out', 'Cancel'],
+      },
+    );
+
+    if (consent.success) {
+      localStorage.removeItem('authToken');
+      await window.electron.removeTokens();
```
Comment (dsasidhar) [inline src/renderer/components/Logout.tsx:20]: Why two places?

File: src/renderer/pages/FileOptions.tsx [hunk 4]
```diff
@@ -323,26 +375,28 @@ export default function FileOptions() {
...
+                }}
               />
               <input
                 className="flex-1 p-2 rounded-md bg-white text-black border-2 border-gray-300"
                 placeholder="Action"
                 value={control.action}
-                onChange={(e) =>
-                  handleControlChange(index, 'action', e.target.value)
-                }
+                onChange={(e) => {
```
Comment (dsasidhar) [inline src/renderer/pages/FileOptions.tsx:392]: good candidate for formik, may be for later. Would add a TODO here.

**Review by dsasidhar — COMMENTED** [review]
Few Comments

File: src/main/main.ts [hunk 5]
```diff
@@ -140,6 +141,34 @@ app.on('window-all-closed', () => {
...
+    if (mainWindow) {
+      if (mainWindow.isMinimized()) mainWindow.restore();
+      mainWindow.focus();
+      const parsedUrl = new URL(url);
+      const accessToken = parsedUrl.searchParams.get('token');
+      const refreshToken = parsedUrl.searchParams.get('refreshToken');
+      if (accessToken && refreshToken) {
+        mainWindow.webContents
+          .executeJavaScript(
+            `localStorage.setItem('authToken', '${accessToken}');`,
```
[the remark this answers is not in the document]
Comment (suraj-turing) [reply src/main/main.ts]: removed localStorage use. now using only local db.

File: src/renderer/App.tsx (hunk 1 above)
Comment (suraj-turing) [reply src/renderer/App.tsx]: implemented

File: src/renderer/App.tsx (hunk 2 above)
Comment (suraj-turing) [reply src/renderer/App.tsx]: done

File: src/renderer/components/Logout.tsx (hunk 3 above)
Comment (suraj-turing) [reply src/renderer/components/Logout.tsx:19]: removed local storage

**Review by dsasidhar — APPROVED** [review]
LGTM

**Pull request merged**